pytest/tests/sanity/congestion_control.py

```diff
@@ -0,0 +1,367 @@
+# This is a test for congestion control. It spins up a single node and loads it
+# with heavy transactions to congest one of the shards. It then checks that the
+# congestion info is correct and that the chain rejected some transactions.
+# Usage:
+# python3 pytest/tests/sanity/congestion_control.py
+
+import unittest
+import pathlib
+import sys
+import json
+import time
+import threading
+
+sys.path.append(str(pathlib.Path(__file__).resolve().parents[2] / 'lib'))
+
+from configured_logger import logger
+from cluster import init_cluster, load_config, spin_up_node, Key
+from utils import load_test_contract, poll_blocks, wait_for_blocks
+from transaction import sign_create_account_with_full_access_key_and_balance_tx, sign_deploy_contract_tx, sign_function_call_tx
+
+ONE_NEAR = 10**24
+TGAS = 10**12
+
+ACCESS_KEY_NONCE_RANGE_MULTIPLIER = 1_000_000
+
+# Shard layout with 5 roughly equal size shards for convenience.
+SHARD_LAYOUT = {
+    "V1": {
+        "boundary_accounts": [
+            "fff",
+            "kkk",
+            "ppp",
+            "uuu",
+        ],
+        "version": 2,
+        "shards_split_map": [],
+        "to_parent_shard_map": [],
+    }
+}
+
+NUM_SHARDS = len(SHARD_LAYOUT["V1"]["boundary_accounts"]) + 1
+
+ACCOUNT_SHARD_0 = "aaa.test0"
+ACCOUNT_SHARD_1 = "ggg.test0"
+ACCOUNT_SHARD_2 = "lll.test0"
+ACCOUNT_SHARD_3 = "rrr.test0"
+ACCOUNT_SHARD_4 = "vvv.test0"
+
+ALL_ACCOUNTS = [
+    ACCOUNT_SHARD_0,
+    ACCOUNT_SHARD_1,
+    ACCOUNT_SHARD_2,
+    ACCOUNT_SHARD_3,
+    ACCOUNT_SHARD_4,
+]
+
+TxHash = str
+AccountId = str
+
+
+class CongestionControlTest(unittest.TestCase):
+
+    def tearDown(self):
+        self.__stop_load()
+
+    def test(self):
+
+        self.nonce = 1
+
+        accounts = self.__prepare_accounts()
+
+        node = self.__setup_node()
+
+        self.__create_accounts(node, accounts)
+
+        self.__deploy_contracts(node, accounts)
+
+        self.__run(node, accounts)
+
+        node.kill()
+
+        logger.info("The test is finished.")
+
+    def __run(self, node, accounts):
+
+        self.__start_load(node, accounts)
+
+        self.__run_under_congestion(node)
+
+        self.__stop_load()
+
+        self.__run_after_congestion(node)
+
+        self.__check_txs(node)
+
+    def __run_under_congestion(self, node):
+        logger.info("Checking the chain under congestion")
+        (start_height, _) = node.get_latest_block()
+        for height, hash in poll_blocks(node, __target=30):
+            # Wait for a few blocks to congest the chain.
+            if height < start_height + 5:
+                continue
+
+            # Check the target shard.
+
+            chunk = self.__get_chunk(node, hash, 0)
+
+            # Check that the target is busy - always using 1000TG.
+            gas_used = chunk['header']['gas_used']
+            self.assertGreaterEqual(gas_used, 1000 * TGAS)
+
+            # Check that the congestion info has no buffered receipts and some deleted receipts.
```

Comment:
probably you meant delayed here? 

```suggestion
            # Check that the congestion info has no buffered receipts and some delayed receipts.
```